databases/week1/homeWork.sql

```diff
@@ -0,0 +1,92 @@
+create DATABASE hyf_db;
+use hyf_db;
+
+CREATE TABLE tasks (
+    id INT AUTO_INCREMENT PRIMARY KEY,
+    title VARCHAR(100),
+    description VARCHAR(255),
+    status VARCHAR(50),
+    due_date DATETIME,
+    created_at DATETIME DEFAULT CURRENT_TIMESTAMP
+);
+INSERT INTO tasks (title, description, status, due_date)
+VALUES ("Finish project report", "Complete the final draft of the quarterly project report.", "NOT_DONE", "2025-02-10"),
+       ("Organize team meeting", "Schedule and prepare for the next team meeting, including an agenda.", "NOT_DONE", "2025-02-12"),
+       ("Review email campaign", "Review and approve the email marketing campaign content.", "NOT_DONE", "2025-02-14"),
+       ("Update website content", "Refresh the home page with new content and images.", "NOT_DONE", "2025-02-16"),
+       ("Submit expense report", "Submit the monthly expense report for approval.", "NOT_DONE", "2025-02-09"),
+       ("Finalize client presentation", "Complete and finalize the presentation for the upcoming client meeting.", "NOT_DONE", "2025-02-11"),
+       ("Team brainstorming session", "Set up a meeting for the team to brainstorm ideas for the new project.", "NOT_DONE", "2025-02-13"),
+       ("Complete training module", "Finish the online training module on new software tools.", "NOT_DONE", "2025-02-17");
+
+       INSERT INTO tasks (title, description, is_done)
+VALUES ("Normalize Database", "database should be normalized by end of this month", FALSE)
+
+
+
+
+
+SELECT * from tasks
+
+SELECT COUNT(title) FROM tasks;  -- Find out how many tasks are in the task table
+
+SELECT * FROM tasks WHERE due_date IS NULL
+
+ALTER TABLE tasks ADD is_done BOOLEAN;
+
+UPDATE tasks
+SET is_done = CASE 
+    WHEN status = "DONE" THEN TRUE  
+    ELSE FALSE
+END;
```

Comment:
Instead of using this `UPDATE` here, you should have used this exact condition (`WHERE status = 'DONE') in the `SELECT` queries. Then there is no need to UPDATE the table.